scripts/upgrade_cfme.py

```diff
@@ -0,0 +1,216 @@
+#!/bin/python
+
+import argparse
+import yaml
+import time
+import threading
+import requests
+from datetime import datetime
+from utils.appliance import IPAppliance
+import logging
+
+# setting script logging object
+# logging all the event to script and progress messages to screen
+formatter = "[%(asctime)s][%(levelname)s][%(funcName)s] - %(message)s"
+log_name = "cfme-upgrade-{date}.log".format(date=datetime.now().strftime("%y:%m:%d::%H:%M:%y"))
+logging.basicConfig(filename=log_name, filemode="w",
+                    level=logging.DEBUG, format=formatter)
+logger = logging.getLogger("cfme_logger")
+
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.INFO)
+console_handler.setFormatter(logging.Formatter(formatter))
+logger.addHandler(console_handler)
+
+
+class CfmeUpgradeManeger(IPAppliance):
+
+    def __init__(self, appliance_ip, repo_list, dest_version):
+        super(CfmeUpgradeManeger, self).__init__(address=appliance_ip)
+        self.repo_list = repo_list
+        self.dest_version = dest_version
+
+    def exec_commaned(self, cmd, expect_failure=False, ignore_failure=False):
+        '''
+        execute command on the appliance and logging the command with it output to log file
+        :param cmd: command
+        :param expect_failure: is the command have to fail
+        :param ignore_failure: not raise exception on command failure
+        :return: command results
+        '''
```

Comment:
We use google style comments: https://google.github.io/styleguide/pyguide.html#Comments